cloud/misc/ovirt_storage.py

```diff
@@ -0,0 +1,432 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: ovirt_storage
+author: "Ryan Groten (@rgroten)"
+short_description: "Create/Delete/Manage VM Disks in oVirt/RHEV"
+description:
+    - Module for managing disks in oVirt/RHEV. Provides the ability to create, attach/detach, or delete both internal (pool) disks to VMs.
+version_added: "2.2"
+requirements:
+    - "ovirt-engine-python-sdk"
+options:
+    state:
+        description:
+            - C(present) will create a new disk and attach it to a VM.
+            - C(absent) will delete a disk attached to a VM.
+            - C(attach) will attach a disk to a VM.
+            - C(detach) will detach a disk from a VM.
+        choices: [ 'present', 'absent', 'attached', 'detached' ]
+        default: present
+    user:
+        description:
+            - User with access to perform disk operations in oVirt.
+        required: true
+    password:
+        description:
+            - Password for I(user).
+        required: true
+    url:
+        description:
+            - URL to oVirt API (U(https://<ovirt-engine>/ovirt-engine/api)).
+        required: true
+    vm_name:
+        description:
+            - Name of the VM in oVirt.
+        required: true
+    disk_alias:
+        description:
+            - Alias of the oVirt disk to perform operation on.
+        required: true
+    disk_size_gb:
+        description:
+            - Size in GB of new disk image. Used when C(state) is C(present).
+    disk_lunid:
+        description:
+            - Create a direct attach fibre channel lun. Must match lunid as the oVirt host sees it. Used when C(state) is C(present).
+    disk_iface:
+        description:
+            - Interface for newly created disk. Used when C(state) is C(present).
+        default: virtio
+        choices: [ 'virtio', 'ide' ]
+    disk_alloc:
+        description:
+            - Allocation Policy for newly created disk. Used when C(state) is C(present).
+        default: cow
+        choices: [ 'cow', 'raw' ]
+    disk_activate:
+        description:
+            - If C(yes), disk will be activated when created or attached to a VM.
+            - If C(no), disk will be attached but not activated.
+            - Used when C(state) is C(present) or C(attached).
+        default: True
+        type: bool
+'''
+
+EXAMPLES = '''
+# Create a new 5GB Pool Disk and attach/activate it to a VM
```

Comment:
@rollandf You're right that's confusing, it probably should be renamed to Image, I will make that change now.